### DISCUSSION TITLE
How do you run mpv with nvidia driver installed?

### DISCUSSION THREAD

**Discussion by BugmanSpirit** [clearfraction/distribution#51] [General]
After installing the proprietary driver, is there some config to tell mpv to use the nvidia driver? Also, what config is recommended?

**Comment by paulcarroty** [comment]
```
hwdec=vdpau
vo=gpu
```

probably enough to use with NVidia GPU.

**Comment by BugmanSpirit** [comment]
I am trying to use an all vulkan config. I built ffmpeg and mpv with vulkan enabled.

But why was vulkan disabled in the first place? Were people facing some issues?

**Reply by paulcarroty** [reply]
It failed to compile with latest Vulkan 1.3 few times, so now disabled.

You can do PR when get the stable build.

**Reply by BugmanSpirit** [reply]
I was able to build it with 1.3.296 but I am not able to get ffmpeg to work with vulkan. I am new to this, I can't tell if my build was improper or my config is messed up. I can see my gpus in vulkanino but ffmpeg says no hardware with vulkan was found
Btw I also have nvidia prime offloading enabled, following arch wiki guide.

**Reply by paulcarroty** [reply]
- https://trac.ffmpeg.org/wiki/HWAccelIntro#Vulkan
- https://ffmpeg.org/ffmpeg-filters.html#Vulkan-Video-Filters

Would be nice to have it, will check later.

**Reply by paulcarroty** [reply]
Update landed, got the next errors:

```
[vo/gpu-next/libplacebo] Failed initializing any SPIR-V compiler! Maybe libplacebo was built without support for either libshaderc or glslang?
[vo/gpu-next/libplacebo] Failed initializing vulkan device
[vo/gpu-next] Failed initializing any suitable GPU context!
```

**Reply by BugmanSpirit** [reply]
This looks like an issue with libplacebo, no? According to my understanding, libplacebo contains mpv algos that can be used by other programs as library. Is it necessary to build ffmpeg with libplacebo? Because mpv would not be relying on ffmpeg to apply shaders/filters that are native to itself.
Latest mpv has disabled the option for enabling shaderc but I do not see any way of telling it to enable glslang either. I'm also unable to enable Spirv-cross. I think issue is with mpv build only.

**Reply by paulcarroty** [reply]
This is the reason: https://github.com/clearlinux-pkgs/libplacebo/blob/main/libplacebo.spec#L105C1-L106C1

**Reply by paulcarroty** [reply]
Replacing [this library](https://download.clearlinux.org/releases/42640/clear/x86_64/os/Packages/libplacebo-lib-7.349.0-82.x86_64.rpm) and extracting [this](https://download.clearlinux.org/current/x86_64/os/Packages/vapoursynth-lib-70.11-3.x86_64.rpm) fixed `mpv --gpu-context=waylandvk`.
